patches/tModLoader/Terraria/ModLoader/Config/ConfigManager.cs

```diff
@@ -72,6 +81,84 @@ internal static void Add(ModConfig config)
 		configList2.Add(GeneratePopulatedClone(config));
 	}
 
+	internal static void FinishSetup()
+	{
+		typesWithLocalizationRegistered.Clear();
+		// Register localization for all fields and properties that should show
+		foreach (var activeConfigs in ConfigManager.Configs) {
+			foreach (var config in activeConfigs.Value) {
+				_ = config.DisplayName;
+
+				RegisterLocalizationKeysForMembers(config.GetType());
+			}
+		}
+	}
+
+	private static void RegisterLocalizationKeysForMembers(Type type)
+	{
+		AssemblyManager.GetAssemblyOwner(type.Assembly, out var modName);
+		foreach (PropertyFieldWrapper variable in ConfigManager.GetFieldsAndProperties(type)) {
+#pragma warning disable CS0618
+			if (Attribute.IsDefined(variable.MemberInfo, typeof(JsonIgnoreAttribute)) && !(Attribute.IsDefined(variable.MemberInfo, typeof(LabelAttribute)) || Attribute.IsDefined(variable.MemberInfo, typeof(ShowDespiteJsonIgnoreAttribute))))
+				continue;
+#pragma warning restore CS0618
+
+			try {
+				RegisterLocalizationKeysForMemberType(variable.Type, type.Assembly);
```

Comment:
not 100% actionable yet, but I believe enums would be better handled in ^. `[Header]` also shouldn't be supported for enums correct? It's just labels for their values.